### PR TITLE
feat: Adds `custom_zone_mapping_zone_id` to `mongodbatlas_global_cluster_config` and deprecates `custom_zone_mapping`

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds `custom_zone_mapping_zone_id` to `mongodbatlas_global_cluster_config` and deprecates `custom_zone_mapping`

Link to any related issue(s): CLOUDP-263795

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/globalclusterconfig/resource_global_cluster_config.go [hunk 1]
```diff
@@ -166,26 +172,36 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
...
+	if err != nil {
+		if httpResp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
 		return diag.FromErr(fmt.Errorf(errorGlobalClusterRead, clusterName, err))
 	}
 
-	if err := d.Set("custom_zone_mapping", globalCluster.GetCustomZoneMapping()); err != nil {
+	if err := d.Set("managed_namespaces", flattenManagedNamespaces(oldResp.GetManagedNamespaces())); err != nil {
```
Comment (AgustinBettati) [inline internal/service/globalclusterconfig/resource_global_cluster_config.go]: nit: should we use the new API for this case? essentially only use the old API value for populating `custom_zone_mapping`

**Review by AgustinBettati — APPROVED** [review]
changes LGTM, would double check on failing CI test

File: internal/service/globalclusterconfig/resource_global_cluster_config.go (hunk 1 above)
Comment (lantoli) [reply internal/service/globalclusterconfig/resource_global_cluster_config.go]: that's a great point, changed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2637/commits/2f7381593625aa54ce212b85cff7e36a82e42b09

File: internal/service/globalclusterconfig/data_source_global_cluster_config.go [hunk 2]
```diff
@@ -54,6 +55,11 @@ func DataSource() *schema.Resource {
 				},
 			},
 			"custom_zone_mapping": {
+				Deprecated: fmt.Sprintf(constant.DeprecationParamByDateWithReplacement, "1.23.0", "custom_zone_mapping_zone_id"),
```
Comment (maastha) [inline internal/service/globalclusterconfig/data_source_global_cluster_config.go]: Should this be updated to use new const similar to `DeprecationParamByVersion`? 
`"This parameter is deprecated and will be removed in version %s. Please transition to %s"`

File: internal/service/globalclusterconfig/resource_global_cluster_config.go [hunk 3]
```diff
@@ -93,6 +94,11 @@ func Resource() *schema.Resource {
 				},
 			},
 			"custom_zone_mapping": {
+				Deprecated: fmt.Sprintf(constant.DeprecationParamByDateWithReplacement, "1.23.0", "custom_zone_mapping_zone_id"),
```
Comment (maastha) [inline internal/service/globalclusterconfig/resource_global_cluster_config.go]: same above https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2637#discussion_r1776788800

File: internal/service/globalclusterconfig/data_source_global_cluster_config.go (hunk 2 above)
Comment (lantoli) [reply internal/service/globalclusterconfig/data_source_global_cluster_config.go]: we were really using that param for versions, so I've changed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2637/commits/a5c40a9437f15380d32824d3d8befe1dd4fbff01

File: internal/service/globalclusterconfig/resource_global_cluster_config.go (hunk 3 above)
Comment (lantoli) [reply internal/service/globalclusterconfig/resource_global_cluster_config.go]: changed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2637/commits/a5c40a9437f15380d32824d3d8befe1dd4fbff01

File: internal/service/globalclusterconfig/data_source_global_cluster_config.go (hunk 2 above)
Comment (maastha) [reply internal/service/globalclusterconfig/data_source_global_cluster_config.go]: oh I see, thanks for updating!

**Review by maastha — APPROVED** [review]

**Review by jwilliams-mongo — APPROVED** [review]

**Pull request merged**